tests/qos/test_pfc_counters.py

```diff
@@ -0,0 +1,144 @@
+from ansible_host import AnsibleHost
+from qos_fixtures import conn_graph_facts
+import os
+import time
+
+"""
+This module implements test cases for PFC counters of SONiC.
+The PFC Rx counter should be increased when the switch receives a priority-based flow control (PFC) pause/unpause frame.
+The PFC Rx counter should NOT be updated when the switch receives a global flow control pause/unpause frame.
+
+In each test case, we send a specific number of pause/unpause frames to a given priority queue of a given port at the
+device under test (DUT). Then we check the SONiC PFC Rx counters. 
+"""
+
+PFC_GEN_FILE_RELATIVE_PATH = r'../../ansible/roles/test/files/helpers/pfc_gen.py'
+""" Expected PFC generator path at the leaf fanout switch """
+PFC_GEN_FILE_DEST = r'~/pfc_gen.py'
+""" Number of generated packets for each test case """
+PKT_COUNT = 10
+""" Number of switch priorities """ 
+PRIO_COUNT = 8
+
+def setup_testbed(ansible_adhoc, testbed, conn_graph_facts):
+    """
+    @Summary: Set up the testbed, including clearing counters, and copying the PFC generator to the leaf fanout switches.
+    @param ansible_adhoc: Fixture provided by the pytest-ansible package. Source of the various device objects. It is
+    mandatory argument for the class constructors.
+    @param testbed: Testbed information
+    @param conn_graph_facts: Testbed topology connectivity information
+    """
+    dut_hostname = testbed['dut']
+    dut_ans = AnsibleHost(ansible_adhoc, dut_hostname)
+    """ Clear PFC counters """
+    dut_ans.sonic_pfc_counters(method = "clear")
+
+    conn_facts = conn_graph_facts['device_conn']	
+    """ Get all the leaf fanout switches """
+    leaf_fanouts = []
+    """ For each interface of DUT"""
+    for intf in conn_facts:
+        peer_device = conn_facts[intf]['peerdevice']
+        if peer_device not in leaf_fanouts:
```

Comment:
Did you mean - if peer_device in leaf_fanouts?

Reply:
Here I want to avoid duplicated leaf fanout switches.

Reply:
@neethajohn I want to avoid duplicated leaf fanout switches